beetsplug/mbweb.py

```diff
@@ -0,0 +1,109 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016, Tigran Kostandyan <t.kostandyan1@gmail.com>
+#
+# Permission is hereby granted, free of charge, to any person obtaining
+# a copy of this software and associated documentation files (the
+# "Software"), to deal in the Software without restriction, including
+# without limitation the rights to use, copy, modify, merge, publish,
+# distribute, sublicense, and/or sell copies of the Software, and to
+# permit persons to whom the Software is furnished to do so, subject to
+# the following conditions:
+#
+# The above copyright notice and this permission notice shall be
+# included in all copies or substantial portions of the Software.
+
+"""Runs a local HTTP server for MusicBrainz web tagger and handles
+users' requests.
+"""
+
+from __future__ import division, absolute_import, print_function
+
+import socket
+import webbrowser
+from io import BytesIO
+from six.moves.urllib_parse import urlparse, urlencode, parse_qs
+from six.moves.BaseHTTPServer import BaseHTTPRequestHandler
+from beets import autotag
+from beets.plugins import BeetsPlugin
+from beets.ui.commands import PromptChoice
+from beets import ui
+
+PORT = 8000
+
+
+# Simple HTTP parser
+class HTTPRequest(BaseHTTPRequestHandler):
+    def __init__(self, request_text):
+        self.rfile = BytesIO(request_text)
+        self.raw_requestline = self.rfile.readline()
+        self.parse_request()
+
+
+# Get 'id' parameter from URI
+def parse(data):
+    r = HTTPRequest(data)
+    parsed = urlparse(r.path)
+    return parse_qs(parsed.query)['id'][0]
+
+
+class Server():
+    def __init__(self):
+        self.host = '127.0.0.1'
+        self.port = PORT
+        try:  # Start TCP socket, catch soket.error
+            self.run_server = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
```

Comment:
I might suggest the name `sock` or something for the socket, for clarity's sake.